kernel/libc/posix/clock_gettime.c

```diff
@@ -86,8 +87,21 @@ int clock_gettime(clockid_t clk_id, struct timespec *ts) {
             ts->tv_nsec = div_result.rem;
             return 0;
 
-        /* Fail out for any other unsupported CPU type */
-        /* case CLOCK_THREAD_CPUTIME_ID: */
+        /* Use the kthread-specific CPU time counters */
+        case CLOCK_THREAD_CPUTIME_ID:
+            /* Check whether they are configured properly
+               as an interval timer. */
+            if(!perf_cntr_timer_enabled()) {
+                errno = EINVAL;
```

Comment:
Just a minor concern. I see that EINVAL is used for the PROCESS CPUTIME ID as well for if perf counters are disabled, but that doesn't directly map to the standard which is that it's used for 'Doesn't specify a known clock'. It would be useful if there was some way to flag the user that the reason they're getting this error is due to a misconfiguration.

Reply:
Do you see any POSIX-complaint way to signal that? This seemed like the most POSIX-y thing to do, but you have a point... Was just going to make it super clear in the Doxygen.

Reply:
Nope! I looked around and couldn't find one myself. That's why I left it as 'minor concern' cause was pretty sure that is just the way to do it and hopefully it doesn't cause any issue.